Documentation/api-guidelines/nullability.md

```diff
@@ -0,0 +1,77 @@
+1. How common is it that an existing override written before the guarantee was put in effect would return null?
+2. How widespread are overrides of the method in question?  This contributes to (1).
+3. How common is it to invoke the method via the base vs via a derived type that may narrow the return type to `T` from `T?`?
+4. How common is it in the case of (3) for such invocations to then dereference the result rather than passing it off to something else that accepts a `T?`?
+
+Object.ToString is arguably the most extreme case.  Answering the above questions:
+1. It is fairly easy in any reasonably-sized code base to find cases, intentional or otherwise, where ToString returns null in some cases (we�ve found examples in corefx, Roslyn, NuGet, ASP.NET, and so on).  One of the most prevalent conditions for this are types that just return the value in a string field which may contain its default value of null, and in particular for structs where a ctor may not have even had a chance to run and validate an input.  Guidance in the docs suggests that ToString shouldn't return null or string.Empty, but even the docs don't follow its own guidance.
+2. Thousands upon thousands of types we don't control override this method today.
+3. It�s common for helper routines to invoke via the base object.ToString, but many ToString uses are actually on derived types.  This is particularly true when working in a code base that both defines a type and consumes its ToString.
+4. Based on examination of several large code bases, we believe it to be relatively rare that the result of an Object.ToString call (made on the base) to be directly dereferenced.  It�s much more common to pass it to another method that accepts `string?`, such as `String.Concat`, `String.Format`, `Console.WriteLine`, logging utilities, and so on.  And while we advocate that ToString results shouldn�t be assumed to be in a particular machine-readable format and parsed, it�s certainly the case that code bases do, such as using `Substring` on the result, but in such cases, the caller needs to understand the format of what�s being rendered, which generally means they�re working with a derived type rather than calling through the base Object.ToString.
+
+As such, for now, we will start with `Object.ToString` returning `string?`.  We can re-evaluate this decision as we get more experience with consumers of the feature.
+
+In contrast, for `Exception.Message` which is also virtual, we plan to have it be non-nullable, even though technically a derived class could override it to return null, because doing so is so rare that we couldn't find any meaningful examples of doing so.
+
+## Code Review Guidance
```

Comment:
Should we add a section for recording the need for potential attributes that allow us record reality better, like the `TryGetValue()` case?